var/spack/repos/builtin/packages/py-pip/package.py

```diff
@@ -113,7 +113,9 @@ def install(self, spec, prefix):
         args = [os.path.join(whl, "pip")] + std_pip_args + ["--prefix=" + prefix, whl]
         python(*args)
 
-    def setup_dependent_package(self, module, dependent_spec):
-        pip = dependent_spec["python"].command
+    def setup_dependent_package(self, module, dependent_spec: Spec):
+        # Old installs do not depend on python-venv, fall back to underlying python.
+        pkg = "python-venv" if dependent_spec.dependencies("python-venv") else "python"
+        pip = dependent_spec[pkg].command
         pip.add_default_arg("-m", "pip")
```

Comment:
this is another place where I think making executables copyable would help:

```suggestion
        pip = python.copy()
        pip.add_default_arg("-m", "pip")
```

Or, along the same lines, if there were a non-mutating `with_default_arg` method:

```suggestion
        pip = python.with_default_arg("-m", "pip")
```

Reply:
Nice, I was going to suggest that above.